src/ir_Mitsubishi.cpp

```diff
@@ -1186,3 +1198,449 @@ String IRMitsubishi136::toString(void) {
+uint8_t IRMitsubishi112::getSwingH(void) {
+  return (remote_state[kMitsubishi112SwingHByte] & kMitsubishi112SwingHMask);
+}
+
+// Emulate a quiet setting. There is no true quiet setting on this a/c
+void IRMitsubishi112::setQuiet(bool on) {
+  if (on)
+    setFan(kMitsubishi112FanQuiet);
+  else if (getQuiet()) setFan(kMitsubishi112FanLow);
+}
+
+// Return the requested power state of the A/C.
+bool IRMitsubishi112::getQuiet(void) {
+  return getFan() == kMitsubishi112FanQuiet;
+}
+
+
+// Convert a standard A/C mode into its native mode.
+uint8_t IRMitsubishi112::convertMode(const stdAc::opmode_t mode) {
+  switch (mode) {
+    case stdAc::opmode_t::kCool: return kMitsubishi112Cool;
+    case stdAc::opmode_t::kHeat: return kMitsubishi112Heat;
+    case stdAc::opmode_t::kDry: return kMitsubishi112Dry;
+    default: return kMitsubishi112Auto;
+  }
+}
+
+// Convert a standard A/C Fan speed into its native fan speed.
+uint8_t IRMitsubishi112::convertFan(const stdAc::fanspeed_t speed) {
+  switch (speed) {
+    case stdAc::fanspeed_t::kMin: return kMitsubishi112FanMin;
+    case stdAc::fanspeed_t::kLow: return kMitsubishi112FanLow;
+    case stdAc::fanspeed_t::kMedium: return kMitsubishi112FanMed;
+    case stdAc::fanspeed_t::kHigh:
+    case stdAc::fanspeed_t::kMax: return kMitsubishi112FanMax;
+    default: return kMitsubishi112FanMed;
+  }
+}
+
+// Convert a standard A/C vertical swing into its native setting.
+uint8_t IRMitsubishi112::convertSwingV(const stdAc::swingv_t position) {
+  switch (position) {
+    case stdAc::swingv_t::kHighest: return kMitsubishi112SwingVHighest;
+    case stdAc::swingv_t::kHigh: return kMitsubishi112SwingVHigh;
+    case stdAc::swingv_t::kMiddle: return kMitsubishi112SwingVMiddle;
+    case stdAc::swingv_t::kLow: return kMitsubishi112SwingVLow;
+    case stdAc::swingv_t::kLowest: return kMitsubishi112SwingVLowest;
+    default: return kMitsubishi112SwingVAuto;
+  }
+}
+
+// Convert a standard A/C vertical swing into its native setting.
+uint8_t IRMitsubishi112::convertSwingH(const stdAc::swingh_t position) {
+  switch (position) {
+    case stdAc::swingh_t::kLeftMax: return kMitsubishi112SwingHLeftMax;
+    case stdAc::swingh_t::kLeft: return kMitsubishi112SwingHLeftInner;
+    case stdAc::swingh_t::kMiddle: return kMitsubishi112SwingHMiddle;
+    case stdAc::swingh_t::kRight: return kMitsubishi112SwingHRightInner;
+    case stdAc::swingh_t::kRightMax: return kMitsubishi112SwingHRightMax;
+    case stdAc::swingh_t::kWide: return kMitsubishi112SwingHWide;
+    case stdAc::swingh_t::kAuto: return kMitsubishi112SwingHAuto;
+    default: return kMitsubishi112SwingHAuto;
+  }
+}
+
+// Convert a native mode to it's common equivalent.
+stdAc::opmode_t IRMitsubishi112::toCommonMode(const uint8_t mode) {
+  switch (mode) {
+    case kMitsubishi112Cool: return stdAc::opmode_t::kCool;
+    case kMitsubishi112Heat: return stdAc::opmode_t::kHeat;
+    case kMitsubishi112Dry: return stdAc::opmode_t::kDry;
```

Comment:
Ditto Fan mode again. If it doesn't have fan only mode, then feel free to ignore these comments/suggestions.